app/api/Files.scala

```diff
@@ -1623,6 +1616,26 @@ class Files @Inject()(
     }
   }
 
+  def bulkDeleteFiles() = PrivateServerAction (parse.json) {implicit request=>
```

Comment:
This action would let anyone logged in call this action and the `removeFile` call doesn't seem to reinforce permissions. We should use something like ```def removeFile(id: UUID) = PermissionAction(Permission.DeleteFile, Some(ResourceRef(ResourceRef.file, id)))``` but I am not sure which permission to use since I don't believe we can pass in a list of resources.

Reply:
Would another option be to files the FileService removeFile method to take permissions into account? It looks like it's supposed to do that, but just doesn't.

Reply:
I made a commit that does that.

Reply:
I implemented the ability to check a list of resources at once a while ago: https://github.com/clowder-framework/clowder/blob/develop/app/api/Permissions.scala#L282 I also added a bulk GET for a list of files: https://github.com/clowder-framework/clowder/blob/develop/app/services/FileService.scala#L104

You can use these in combination like this:
https://github.com/clowder-framework/clowder/blob/develop/app/util/SearchUtils.scala#L302

both of these calls return little objects that tell you what does/doesn't have permission and what was/wasn't found (if you asked for multiple files). Please use this pattern instead. The changes here, you added a new permission check inside the file service but existing uses of that call in api Files have already done the check so now it is checking twice. 
If you remove the permission check in the service, and use the pattern above inside bulkDeleteFiles, it will use existing code and be a bit more efficient.

Reply:
@max-zilla thanks. that will be much better. i'll get this changed and push a new commit later today.

Reply:
@max-zilla I made the change. Let me know if it looks okay.

Reply:
@tcnichol @max-zilla not sure about the changes to https://github.com/clowder-framework/clowder/blob/develop/app/services/mongodb/MongoDBFileService.scala#L818. I don't see are reference to permissions and the changes in the branch just look like spacing changes? Am I looking in the wrong place? Thank you.

Reply:
That commit is just spacing changes.

The actual change in checking permissions is in this file:

https://github.com/clowder-framework/clowder/pull/153/files'

line 1627 uses the new checkPermission method that takes in a list of resourceRef. I then removed any permission checks elsewhere.

Reply:
It looks like I removed the redundant check he mentioned a few commits ago on this one.